presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -347,6 +348,21 @@ public static OrcSelectivePageSource createOrcPageSource(
                             hiveColumnIndex -> new VariableReferenceExpression(columnNames.get(hiveColumnIndex), columnTypes.get(hiveColumnIndex)),
                             hiveColumnIndex -> new InputReferenceExpression(inputs.get(hiveColumnIndex), columnTypes.get(hiveColumnIndex))));
 
+            List<HiveColumnHandle> preCoercedColumHandles = physicalColumns
+                    .stream()
+                    .filter((hiveColumnHandle -> hiveColumnHandle.getPreCoersedType().isPresent()))
+                    .collect(Collectors.toList());
+
+            ImmutableMap<VariableReferenceExpression, InputReferenceExpression> preCoercedTypes = preCoercedColumHandles
+                    .stream()
+                    .collect(toImmutableMap(
+                            columnHandle -> new VariableReferenceExpression(columnHandle.getName(), typeManager.getType(columnHandle.getPreCoersedType().get())),
+                            columnHandle -> new InputReferenceExpression(inputs.get(columnHandle.getHiveColumnIndex()), columnTypes.get(columnHandle.getHiveColumnIndex()))));
+
```

Comment:
An alternate way is to get rid of streams and then rely on `ifPresent` so that you don't have to filter and collect.

```
            ImmutableMap.Builder<VariableReferenceExpression, InputReferenceExpression> builder = ImmutableMap.builder();
            for (HiveColumnHandle hiveColumnHandle : physicalColumns)
            {
                hiveColumnHandle.getPreCoersedType().ifPresent(
                        type -> builder.put(
                                new VariableReferenceExpression(
                                        hiveColumnHandle.getName(),
                                        typeManager.getType(type)),
                                new InputReferenceExpression(
                                        inputs.get(hiveColumnHandle.getHiveColumnIndex()),
                                        columnTypes.get(hiveColumnHandle.getHiveColumnIndex()))
                        )
                );
            }
```